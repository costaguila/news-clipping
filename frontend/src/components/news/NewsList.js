import React, { Component } from 'react';
import axios from 'axios';

import NewsItem from './News_item';
import SearchBar from '../searchbar/SearchBar';

import { getNews, getNewsSource } from '../../services/newsService';

class NewsList extends Component {

    constructor(props){
        super(props)
        let fontes = [];
        let noticias = [];
        this.state = {
            fontes: fontes,
            fontesFiltradas: fontes,
            itens: noticias,
            itensFiltrados: noticias,
            searchbar: props.searchbar ? props.searchbar : true
        }
    }

    componentDidMount(){
      axios.all([getNews(),getNewsSource()]).then(axios.spread((noticias,fontes)=>{
        this.setState({itens:noticias,itensFiltrados:noticias, fontes:fontes, fontesFiltradas:fontes})
      }))
    }

    getFonteName(title){
        return title ? title : 'Nao informado'
    }

    filterListas(name) {
        if (name !== ""){
            let filtro = this.state.fontes.filter(x => x.name.toUpperCase().includes(name.toUpperCase()));
            this.setState( {fontesFiltradas: filtro} );
        }else{
            this.setState( {fontesFiltradas: this.state.fontes} );
        }
    }

    filterNome(name) {
        if (name !== ""){
            let filtro = this.state.itens.filter(x => x.title.toUpperCase().includes(name.toUpperCase()));
            this.setState( {itensFiltrados: filtro} );
        }else{
            this.setState( {itensFiltrados: this.state.itens} );
        }
    }

    render(){
        const listaNoticiasPorFonte = this.state.fontesFiltradas.map(fonte =>(
          <div className="newsSource" key={fonte.name} >
              <div className="newssourceHeader">
                <h1># {this.getFonteName(fonte.name)}</h1>
              </div>
              <div className="newssourceContainer">
                  {
                    this.state.itensFiltrados.filter(noticias => noticias.spider == fonte.name)
                    .map(item => <NewsItem key={item.link} item={item}/> )
                  }
              </div>
          </div>
        ))

        return (
            <div key='listaDeNoticias'>
                <SearchBar  legenda='Busca por fonte' filterListas={this.filterListas.bind(this)} />
                <SearchBar  legenda='Busca por titulo' filterListas={this.filterNome.bind(this)} />
                {listaNoticiasPorFonte}
            </div>
        )
    }
}

export default NewsList;
