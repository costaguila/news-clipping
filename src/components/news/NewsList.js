import React, { Component } from 'react';
import NewsItem from './News_item';
import { getNewsById } from '../../services/newsService';

class NewsList extends Component {

    constructor(props){
        super(props)
        this.state = {
            fonte: props.fonteNoticias,
            itens: getNewsById(props.fonteNoticias.id),
            name: props.fonteNoticias.title ? props.fonteNoticias.title : 'Nao informado'
        }
    }

    render(){
        return (
            <div className="newsSource">
                <div className="newssourceHeader"><h1># {this.state.name}</h1></div>
                <div className="newssourceContainer">
                    {
                    this.state.itens.map(selectedItem=>(<NewsItem key={this.state.id} item={ selectedItem } />))
                    }
                </div>
            </div>
        );
    }
}

export default NewsList;
