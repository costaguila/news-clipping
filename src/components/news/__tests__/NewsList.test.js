import React from 'react';
import { render } from '../../../setupTests';

import NewsList from '../NewsList';

describe('<NewsList />', () => {
  it('Deve possuir um titulo # g1', () => {
    //arrange
    const fonteNoticiasObj = {'id': 0,'title':'g1'}
    //act
    const component = render(<NewsList debug  key={fonteNoticiasObj.id} fonteNoticias={fonteNoticiasObj}/>);
    let header = component.find('.newssourceHeader h1')
    //assert
    expect(header[0]['children'][0].data).toEqual('# g1')
    expect(component).toMatchSnapshot();
  });
  it('1 + 1 = 2', () => {
    //arrange
    let result = 0
    //act
    result = 1+1
    //assert
    expect(result).toMatchSnapshot(2);
  });
});
