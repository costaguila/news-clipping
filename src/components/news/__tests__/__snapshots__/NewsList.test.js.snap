// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsList /> 1 + 1 = 2 1`] = `2`;

exports[`<NewsList /> Deve possuir um titulo # g1 1`] = `
<div
  class="newsSource"
>
  <div
    class="newssourceHeader"
  >
    <h1>
      # g1
    </h1>
  </div>
  <div
    class="newssourceContainer"
  >
    <div
      class="newsItem"
    >
      <img
        alt="Imagem nao fornecida"
        src="royalty-free-music-videos.jpg"
      />
      <p>
        Guerra na síria
      </p>
    </div>
    <div
      class="newsItem"
    >
      <img
        alt="Imagem nao fornecida"
        src="royalty-free-music-videos.jpg"
      />
      <p>
        José falou com a ana
      </p>
    </div>
  </div>
</div>
`;
